funcx_endpoint/funcx_endpoint/endpoint/interchange.py

```diff
@@ -367,35 +375,68 @@ def _command_server(self, kill_event):
                 # logger.debug("[COMMAND] is alive")
                 continue
 
+    def quiesce(self):
+        """Temporarily stop everything on the interchange in order to reach a consistent
+        state before attempting to start again. This must be called on the main thread
+        """
+        logger.info("Interchange Quiesce in progress (stopping and joining all threads)")
+        self._quiesce_event.set()
+        self._task_puller_thread.join()
+        self._command_thread.join()
+        # this must be called last to ensure the next interchange run will occur
+        self._quiesce_event.clear()
+
     def stop(self):
         """Prepare the interchange for shutdown"""
```

Comment:
@yadudoc I went with `stop` rather than `shutdown` as that is what was used here before and what is used in executor interchanges. We can switch to `shutdown` if you think it makes more sense